piet-gpu-hal/src/vulkan.rs

```diff
@@ -70,18 +124,74 @@ impl VkInstance {
         unsafe {
             let app_name = CString::new("VkToy").unwrap();
             let entry = Entry::new()?;
+
+            let exist_lyrs = entry
+                .enumerate_instance_layer_properties()?;
+            let lyrs = LAYERS.iter().filter_map(|&lyr| {
+                exist_lyrs
+                    .iter()
+                    .find(|x|
+                        CStr::from_ptr(x.layer_name.as_ptr()) == lyr
+                    )
+                    .map(|_| lyr.as_ptr())
+                    .or_else(|| {
+                        println!("Unable to layer: {}, have you installed the Vulkan SDK?", lyr.to_string_lossy());
+                        None
+                    })
+            }).collect::<Vec<*const i8>>();
```

Comment:
The type is `std::os::raw::c_char`, which is not necessarily the same as `i8` - this may cause a type error on platforms that have unsigned char. In this particular case you can write `collect::<Vec<_>>()` which I think is idiomatic.